### ISSUE TITLE
Nine more rules write only one of the two spellings of a selector carrying an inline comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#193]
Found while fixing [#190](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/190), whose parting note asked for this sweep. The same read-raw-write-raw shape stands in nine more rules — seven of their own and two through the checker they share — and every one of them drops its fix in the same silence on a selector carrying an inline comment under `postcss-scss` — reported, fixed, and the file untouched, which is the failure mode of [#126](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/126).

### What minimal example or steps are needed to reproduce the bug?

One shape per rule, each with the configuration named beside it, each linted with `--fix` under `customSyntax: "postcss-scss"`:

| rule | option | code | expected fix, which never arrives |
|---|---|---|---|
| `selector-pseudo-class-case` | `"lower"` | `.a // c`⏎`b:HOVER {}` | `b:hover` |
| `selector-list-comma-space-after` | `"always"` | `.a // c`⏎`.b,.c {}` | `.b, .c` |
| `selector-list-comma-space-before` | `"never"` | `.a // c`⏎`.b , .c {}` | `.b, .c` |
| `selector-list-comma-newline-after` | `"always"` | `.a // c`⏎`.b, .c {}` | a break after the comma |
| `selector-list-comma-newline-before` | `"never-multi-line"` | `.a // c`⏎`.b`⏎`, .c {}` | the break before the comma taken away |
| `selector-max-empty-lines` | `0` | `.a // c`⏎⏎⏎`.b {}` | the empty lines closed up |
| `no-eol-whitespace` | `true` | `.a // c `⏎`.b {}` | the trailing space taken away |
| `selector-attribute-operator-space-before` | `"never"` | `.a // c`⏎`[t ='y'] {}` | `[t='y']` |
| `selector-attribute-operator-space-after` | `"always"` | `.a // c`⏎`[t='y'] {}` | `[t= 'y']` |

In every case the warning is reported, `--fix` reports a clean pass, the file comes back byte-identical, and a relint reports the same warning again.

### What is the cause?

The same one #190 names for `selector-attribute-brackets-space-inside`. `postcss-scss` keeps two spellings of such a selector — the raw the rules read, with every `//` comment spelled as a block one, and `raws.selector.scss`, the file's own spelling, which is the copy the stringifier prints. Each of the seven reads the raw and writes its fix back to the raw alone — the operator pair through `selectorAttributeOperatorSpaceChecker`, which does the same:

```js
let selector = ruleNode.raws.selector ? ruleNode.raws.selector.raw : ruleNode.selector
…
if (ruleNode.raws.selector) ruleNode.raws.selector.raw = fixedSelector
```

so the printed copy never learns of the fix, and the two copies are left out of step for whatever reads the pair next — the corruption half of #190 grows from exactly that. The positions of the warnings are counted in the raw as well, so they drift two characters per comment from the file.

### Do you have a proposal to fix the bug?

The write-back and the translation that #190 gave the brackets rule, which `selector-descendant-combinator-no-non-space`, `selector-pseudo-class-parentheses-space-inside` and `string-quotes` already share: read the pair through `findSelectorInlineComments`, send every reported position through `toSelectorSourceIndex`, and on a fix write `raws.selector.raw` and, where the `scss` half stands, `restoreSelectorInlineComments` into it.

`no-eol-whitespace` is the odd one of the nine: it edits the selector as text rather than through a parse, and the trailing whitespace it takes away may stand inside the comment's own line, so the restore-by-ordinal machinery fits it less directly and it may want a reading of its own.

This sweep covered every rule and checker that writes `raws.selector.raw` today. `indentation` writes `node.selector` directly and fails another way — it destroys the comment outright — which is its own issue.
